Sources/Moya/Plugins/AccessTokenPlugin.swift

```diff
@@ -6,47 +6,49 @@ import Result
 /// A protocol for controlling the behavior of `AccessTokenPlugin`.
 public protocol AccessTokenAuthorizable {
 
-    /// Represents the authorization header to use for requests.
+    /// Represents the authorization header or url parameter to use for requests.
     var authorizationType: AuthorizationType { get }
 }
 
 // MARK: - AuthorizationType
 
-/// An enum representing the header to use with an `AccessTokenPlugin`
+/// An enum representing the header or url parameter to use with an `AccessTokenPlugin`
 public enum AuthorizationType: String {
     case none
     case basic = "Basic"
     case bearer = "Bearer"
+    case urlParameter = "access_token"
```

Comment:
I am thinking this might not cover everything now. What if you have an API that uses `api_key` or `api_token` as an url parameter?

Reply:
I had similar thoughts regarding how many cases this plugin would support. I was thinking about changing the enum to use associated values.
```Swift
enum AuthorizationType {
    case header(String)
    case urlParameter(String)
    case none
}
```

Reply:
If you needed to use the header and urlParameters you would have to add the parameters in the TargetType still. The AccessTokenPlugin was designed around supplying header values.

Reply:
I feel like it would make sense to support a way to add a specific parameter to every request in a provider though.

Reply:
What if we had a `ParameterPlugin` or `HeaderPlugin` that would add a shared parameter or header across `TargetTypes`? Too awkward? I agree with the need for reuse.

Reply:
Hello guys, you suggested nice ideas. Can I try to implement case with argumented enum or additional plugins?

Reply:
I agree we could improve current plugin to allow using the token as a parameter (I had multiple occasions to use it once as a parameter and once as a header). 

How about adding the `parameter` name in initializer? We could also add `header` name if needed. Then the cases could be:
```swift

init(tokenClosure: ..., parameterName: ..., headerName: ...) { ... }

enum AuthorizationType {
    case basicInHeaders
    case bearerInHeaders
    case basicInParameters
    case bearerInParameters
    none
}
```

Just an idea, though.

Reply:
@dotprox I don't know what's best to be honest and it's not for me to decide alone. I'm just trying to drive some discussion. I don't want to ask you to do A then decide B 😐 Maybe we should open an issue discussing reuse across multiple providers and see if anyone has other ideas/input

Reply:
Guys, what you think about combination of optional name for token and initialization with token key? If you need to use prefix, you can add specific name, otherwise you will use default names like `access_token` or `Authorization`.

```swift
init(tokenClosure: ..., parameterName: ... = "access_token", headerName: ... = "Authorization") { ... }

public enum AuthorizationType: String {
    case none
    case header(name: String?) // -> Authorization: <name?><token>
    case parameter(name: String?) // -> access_token=<name?><token>
}
```

inspired by @sunshinejr and @SD10 suggestions

Reply:
```Swift
public struct HeaderPlugin: PluginType {
    var headerValue: String
    var headerName: String
    var shouldApply: Bool
}
public struct ParameterPlugin: PluginType {
   var parameterValue: Any
   var parameterName: String
   var encoding: ParameterEncoding
   var shouldApply: Bool
}
```
This sort of removes the concept of auth but it is a way to share a specific header or parameter across multiple providers. @sunshinejr I like what you suggested but what feels awkward is when I don't want to deal with a `header` vs a `parameter` or vice versa. It would be nice to see this decoupled.

Reply:
Hmm, how would it handle having one call with `parameter` and one with `header` in one `TargetType`? Also, I agree that we could discuss it in another issue as this one will be buried fast.

Reply:
```Swift
protocol ParameterApplicable {
    var shouldApply: Bool
}
protocol HeaderApplicable {
    var shouldApply: Bool
}
```
The `TargetType` would need to conform to one or both of these then you just return `true` or `false` depending on the `case` of the target. I can open an issue a little bit later documenting what's been discussed

Reply:
:+1: On moving the discussion to an issue

Reply:
Created an issue #1283 to summarize what has been discussed here